pkgs/development/tools/xcbuild/platform.nix

```diff
@@ -0,0 +1,170 @@
+{ stdenv, sdk, writeText, platformName, xcbuild }:
+
+let
+
+  Info = {
+    CFBundleIdentifier = platformName;
+    Type = "Platform";
+    Name = "macosx";
+  };
+
+  Version = {
+    ProjectName = "OSXPlatformSupport";
+  };
+
+  Tools = [
+    {
+      Identifier = "com.apple.build-tools.nmedit";
+      Type = "Tool";
+      Name = "Nmedit";
+    }
+    {
+      Identifier = "com.apple.compilers.resource-copier";
+      Type = "Tool";
+      Name = "Resource Copier";
+    }
+    {
+      Identifier = "com.apple.compilers.yacc";
+      Type = "Tool";
+      Name = "Yacc";
+      InputFileTypes = [ "sourcecode.yacc" ];
+      ExecDescription = "Yacc $(InputFile)";
+    }
+    {
+      Identifier = "com.apple.compilers.lex";
+      Type = "Tool";
+      Name = "Lex";
+      ExecDescription = "Lex $(InputFile)";
+      InputFileTypes = [ "sourcecode.lex" ];
+    }
+  ];
+
+  Architectures = [
+    {
+		  Identifier = "Standard";
+	    Type = "Architecture";
+		  Name = "Standard Architectures (64-bit Intel)";
+		  RealArchitectures = [ "x86_64" ];
+      ArchitectureSetting = "ARCHS_STANDARD";
+    }
+    {
+      Identifier = "Universal";
+      Type = "Architecture";
+      Name = "Universal (64-bit Intel)";
+      RealArchitectures = [ "x86_64" ];
+      ArchitectureSetting = "ARCHS_STANDARD_32_64_BIT";
+    }
+    {
+      Identifier = "Native";
+      Type = "Architecture";
+      Name = "Native Architecture of Build Machine";
+      ArchitectureSetting = "NATIVE_ARCH_ACTUAL";
+    }
+    {
+      Identifier = "Standard64bit";
+      Type = "Architecture";
+      Name = "64-bit Intel";
+      RealArchitectures = [ "x86_64" ];
+      ArchitectureSetting = "ARCHS_STANDARD_64_BIT";
+    }
+    {
+      Identifier = "x86_64";
+      Type = "Architecture";
+      Name = "Intel 64-bit";
+    }
+    {
+      Identifier = "Standard_Including_64_bit";
+      Type = "Architecture";
+      Name = "Standard Architectures (including 64-bit)";
+      RealArchitectures = [ "x86_64" ];
+		  ArchitectureSetting = "ARCHS_STANDARD_INCLUDING_64_BIT";
+    }
+  ];
+
+  PackageTypes = [
```

Comment:
I'm not necessarily opposed to putting this stuff in here, but could you add some comments on this nix-plist stuff explaining whether it differs from upstream in meaningful ways, or in general how customized it is?

I'm thinking a few months down the line when @grp has made improvements upstream and we want to pull more stuff in from `xcbuild` and I'll be unsure whether this stuff is picked from upstream with minor changes, or hand-written from scratch for our purposes, or is literally identical to upstream except in Nix rather than plists, etc.

Does that make sense?